### DISCUSSION TITLE
PG JDBC binary data transfer

### DISCUSSION THREAD

**Discussion by asinitsyn** [questdb/questdb#4958] [Ideas]
I was profiling application that queries a lot of data from QuestDB using postgres JDBC driver and noticed that all the data is returned as  Strings and then converted to long or double by JDBC driver. It creates a lot of garbage on the client side to parse from strings. It would be great to avoid binary -> String -> binary transformation. I am sure it will have positive impact on speed of queries.

**Comment by tris0laris** [comment]
Thanks for reaching out - a binary protocol will soon be in the works. We'll keep you posted and will track this item in our roadmap once we start working on it.
